pkg/end_tests/two_player_game_test/two_player_game_test.go

```diff
@@ -0,0 +1,705 @@
+		Game:        game,
+		Position:    pos,
+		Side:        side.Left,
+		FeatureType: feature.Road,
+		MeepleExist: true,
+		TurnNumber:  2,
+	}.Run()
+
+	test.CheckMeeplesAndScore{
+		Game:          game,
+		TestingT:      t,
+		PlayerScores:  []uint32{4, 0},
+		PlayerMeeples: []uint8{7, 6},
+		TurnNumber:    2,
+	}.Run()
+}
+
+// road turn
+// player 1 places meeple (m) on a field
+/*
+|       0  1
+|
+| 	   ......
+|1 	   -1-M2.
+| 	   / \ |.
+| 	   \ /m|.
+|0 	   -0--3.
+| 	   ......
+*/
+func checkThirdTurn(game *gameMod.Game, t *testing.T) {
+	pos := position.New(1, 0)
+
+	// try illegal turn first (put meeple on a road)
+	test.MakeWrongTurn{
+		Game:         game,
+		TestingT:     t,
+		TilePosition: pos,
+		MeepleParams: test.MeepleParams{MeepleType: elements.NormalMeeple, FeatureSide: side.Top, FeatureType: feature.Road},
+		TurnNumber:   3,
+	}.Run()
+
+	// normal correct turn
+	test.MakeTurn{
+		Game:         game,
+		TestingT:     t,
+		TilePosition: pos,
+		MeepleParams: test.MeepleParams{MeepleType: elements.NormalMeeple, FeatureSide: side.TopLeftEdge, FeatureType: feature.Field},
+		TurnNumber:   3,
+	}.Run()
+
+	test.VerifyMeepleExistence{
+		TestingT:    t,
+		Game:        game,
+		Position:    pos,
+		Side:        side.TopLeftEdge,
+		FeatureType: feature.Field,
+		MeepleExist: true,
+		TurnNumber:  3,
+	}.Run()
+
+	test.CheckMeeplesAndScore{
+		Game:          game,
+		TestingT:      t,
+		PlayerScores:  []uint32{4, 0},
+		PlayerMeeples: []uint8{6, 6},
+		TurnNumber:    3,
+	}.Run()
+}
+
+// T cross road
+// player2 places meeple on road going down
+/*
+|   -1  0  1
+|
+| 	   ......
+|1 	   -1-M2.
+| 	   / \ |.
+| 	.| \ /m|.
+|0 	.4--0--3.
+| 	.M.......
+*/
+func checkFourthTurn(game *gameMod.Game, t *testing.T) {
+	pos := position.New(-1, 0)
+
+	// try illegal turn first (put meeple on a road)
+	test.MakeWrongTurn{
+		Game:         game,
+		TestingT:     t,
+		TilePosition: pos,
+		MeepleParams: test.MeepleParams{MeepleType: elements.NormalMeeple, FeatureSide: side.Right, FeatureType: feature.Road},
+		TurnNumber:   4,
+	}.Run()
+
+	// normal correct turn
+	test.MakeTurn{
+		Game:         game,
+		TestingT:     t,
+		TilePosition: pos,
+		MeepleParams: test.MeepleParams{MeepleType: elements.NormalMeeple, FeatureSide: side.Bottom, FeatureType: feature.Road},
+		TurnNumber:   4,
+	}.Run()
+
+	test.VerifyMeepleExistence{
+		TestingT:    t,
+		Game:        game,
+		Position:    pos,
+		Side:        side.Bottom,
+		FeatureType: feature.Road,
+		MeepleExist: true,
+		TurnNumber:  4,
+	}.Run()
+
+	test.CheckMeeplesAndScore{
+		Game:          game,
+		TestingT:      t,
+		PlayerScores:  []uint32{4, 0},
+		PlayerMeeples: []uint8{6, 5},
+		TurnNumber:    4,
+	}.Run()
+}
+
+// monastery with single road
+// player1 places meeple on a monastery
+// road from 4 to 5 is finished, so player2 scores 2 points
+/*
+|   -1  0  1
+|
+| 	   ......
+|1 	   -1-M2.
+| 	   / \ |.
+| 	.| \ /m|.
+|0 	.4--0--3.
+| 	.|.......
+|	.|.
+|-1 .5.
+| 	...
```

Comment:
The meeple on monastery doesn't seem to be indicated in any way?

Reply:
because there is no space for meeple in 3x3 ascii drawings. Would you like me to convert all drawings in that to 5x5 size? (this was my first ever test, and thus it was 3x3. After that i started drawing 5x5)

Reply:
Ideally, these would be in 5x5 size, yeah.